src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -81,16 +84,16 @@ class PosHashTable {
   // of the offset does not limit the number of rows in the partition but the number of distinct values. If we end up
   // with a partition that has more values, the partitioning algorithm is at fault.
   using Offset = uint32_t;
-  using HashTable = ska::bytell_hash_map<HashedType, Offset>;
+  using HashTable = tsl::robin_pg_map<HashedType, Offset, std::hash<HashedType>, std::equal_to<HashedType>, PolymorphicAllocator<std::pair<HashedType, Offset>>>;
 
   // The small_vector holds the first n values in local storage and only resorts to heap storage after that. 1 is chosen
   // as n because in many cases, we join on primary key attributes where by definition we have only one match on the
   // smaller side.
-  using SmallPosList = boost::container::small_vector<RowID, 1>;
+  using SmallPosList = boost::container::small_vector<RowID, 1, PolymorphicAllocator<RowID>>; // TODO PMR raus
```

Comment:
(2) No, PMR is useful. But: check if small_vector is still beneficial over std::vector?

Reply:
https://github.com/hyrise/hyrise/compare/b29c422..98d8c78

small_vector is still better than std::vector

```
+----------++----------+---------+--------++----------+----------+--------+---------+
| Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
|          ||      old |     new |        ||      old |      new |        |         |
+----------++----------+---------+--------++----------+----------+--------+---------+
| TPC-H 01 ||   1668.3 |  1750.0 |   +5%  ||     0.60 |     0.57 |   -5%  |  0.0103 |
| TPC-H 02 ||     12.0 |    12.7 |   +6%  ||    83.60 |    78.52 |   -6%  |  0.0000 |
| TPC-H 03 ||    182.3 |   182.5 |   +0%  ||     5.49 |     5.48 |   -0%  |  0.5887 |
| TPC-H 04 ||    156.4 |   154.2 |   -1%  ||     6.39 |     6.49 |   +1%  |  0.0000 |
| TPC-H 05 ||    488.4 |   501.8 |   +3%  ||     2.05 |     1.99 |   -3%  |  0.0000 |
| TPC-H 06 ||      7.7 |     8.2 |   +6%  ||   129.13 |   122.09 |   -5%  |  0.0000 |
| TPC-H 07 ||    132.4 |   140.6 |   +6%  ||     7.55 |     7.11 |   -6%  |  0.0000 |
| TPC-H 08 ||    159.6 |   165.8 |   +4%  ||     6.27 |     6.03 |   -4%  |  0.0000 |
| TPC-H 09 ||   1021.5 |  1060.4 |   +4%  ||     0.98 |     0.94 |   -4%  |  0.0000 |
| TPC-H 10 ||    516.7 |   529.8 |   +3%  ||     1.94 |     1.89 |   -2%  |  0.0179 |
| TPC-H 11 ||     16.5 |    17.6 |   +7%  ||    60.74 |    56.91 |   -6%  |  0.0000 |
| TPC-H 12 ||     90.5 |    99.3 |  +10%  ||    11.04 |    10.07 |   -9%  |  0.0000 |
| TPC-H 13 ||    727.6 |   769.1 |   +6%  ||     1.37 |     1.30 |   -5%  |  0.0000 |
| TPC-H 14 ||     39.8 |    43.2 |   +8%  ||    25.10 |    23.14 |   -8%  |  0.0000 |
| TPC-H 15 ||     17.6 |    18.6 |   +6%  ||    56.92 |    53.80 |   -5%  |  0.0000 |
| TPC-H 16 ||    179.7 |   194.5 |   +8%  ||     5.56 |     5.14 |   -8%  |  0.0000 |
| TPC-H 17 ||     54.5 |    55.4 |   +2%  ||    18.35 |    18.05 |   -2%  |  0.0000 |
| TPC-H 18 ||   1301.4 |  1365.5 |   +5%  ||     0.77 |     0.73 |   -5%  |  0.0000 |
| TPC-H 19 ||     75.8 |    79.0 |   +4%  ||    13.20 |    12.66 |   -4%  |  0.0000 |
| TPC-H 20 ||     32.1 |    33.6 |   +5%  ||    31.16 |    29.74 |   -5%  |  0.0000 |
| TPC-H 21 ||    845.6 |   893.2 |   +6%  ||     1.18 |     1.12 |   -5%  |  0.0000 |
| TPC-H 22 ||    134.7 |   134.8 |   +0%  ||     7.42 |     7.42 |   -0%  |  0.9508 |
+----------++----------+---------+--------++----------+----------+--------+---------+
| Sum      ||   7861.0 |  8209.6 |   +4%  ||          |          |        |         |
| Geomean  ||          |         |        ||          |          |   -4%  |         |
+----------++----------+---------+--------++----------+----------+--------+---------+
```

Reply:
Weird. It's minimally better on nemea. Still, if it has a negative effect in your measurements and a negligible effect here, let's keep the small_vector.

```diff

 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |     new |        ||      old |      new |        |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | TPC-H 01 ||    812.8 |   831.1 |   +2%  ||     1.23 |     1.20 |   -2%  |  0.0011 |
 | TPC-H 02 ||      4.5 |     4.5 |   -0%  ||   220.02 |   220.06 |   +0%  |  0.9815 |
 | TPC-H 03 ||     81.1 |    81.2 |   +0%  ||    12.33 |    12.31 |   -0%  |  0.2183 |
 | TPC-H 04 ||     65.7 |    65.8 |   +0%  ||    15.23 |    15.19 |   -0%  |  0.0004 |
 | TPC-H 05 ||    220.1 |   226.0 |   +3%  ||     4.54 |     4.42 |   -3%  |  0.0000 |
 | TPC-H 06 ||      3.2 |     3.2 |   +0%  ||   312.96 |   312.77 |   -0%  |  0.5545 |
 | TPC-H 07 ||     57.1 |    56.8 |   -0%  ||    17.53 |    17.59 |   +0%  |  0.6873 |
 | TPC-H 08 ||     72.0 |    71.9 |   -0%  ||    13.88 |    13.91 |   +0%  |  0.6791 |
 | TPC-H 09 ||    506.3 |   507.2 |   +0%  ||     1.98 |     1.97 |   -0%  |  0.6233 |
 | TPC-H 10 ||    248.0 |   250.7 |   +1%  ||     4.03 |     3.99 |   -1%  |  0.2820 |
 | TPC-H 11 ||      8.2 |     8.2 |   +1%  ||   122.12 |   121.19 |   -1%  |  0.0000 |
 | TPC-H 12 ||     42.2 |    42.2 |   -0%  ||    23.68 |    23.71 |   +0%  |  0.1887 |
 | TPC-H 13 ||    351.2 |   350.9 |   -0%  ||     2.85 |     2.85 |   +0%  |  0.6242 |
 | TPC-H 14 ||     21.0 |    20.9 |   -0%  ||    47.72 |    47.93 |   +0%  |  0.0000 |
 | TPC-H 15 ||      7.7 |     7.7 |   +0%  ||   130.40 |   130.24 |   -0%  |  0.0627 |
 | TPC-H 16 ||     92.5 |    90.5 |   -2%  ||    10.81 |    11.05 |   +2%  |  0.0000 |
 | TPC-H 17 ||     16.3 |    16.3 |   +0%  ||    61.23 |    61.21 |   -0%  |  0.8515 |
+| TPC-H 18 ||    718.3 |   659.6 |   -8%  ||     1.39 |     1.52 |   +9%  |  0.0000 |
 | TPC-H 19 ||     27.7 |    27.8 |   +0%  ||    36.11 |    36.02 |   -0%  |  0.0001 |
 | TPC-H 20 ||     15.6 |    15.6 |   -0%  ||    64.18 |    64.25 |   +0%  |  0.6501 |
 | TPC-H 21 ||    412.2 |   425.0 |   +3%  ||     2.43 |     2.35 |   -3%  |  0.0000 |
 | TPC-H 22 ||     49.4 |    49.3 |   -0%  ||    20.26 |    20.29 |   +0%  |  0.0006 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Sum      ||   3832.9 |  3812.4 |   -1%  ||          |          |        |         |
 | Geomean  ||          |         |        ||          |          |   +0%  |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
```

Reply:
Same on nemea:

```diff
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |     new |        ||      old |      new |        |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | TPC-H 01 ||    824.4 |   812.6 |   -1%  ||     1.21 |     1.23 |   +1%  |  0.0224 |
-| TPC-H 02 ||      4.9 |     5.2 |   +6%  ||   205.76 |   193.44 |   -6%  |  0.0000 |
 | TPC-H 03 ||     84.6 |    84.4 |   -0%  ||    11.82 |    11.85 |   +0%  |  0.0138 |
 | TPC-H 04 ||     68.6 |    69.2 |   +1%  ||    14.57 |    14.46 |   -1%  |  0.0000 |
 | TPC-H 05 ||    229.6 |   233.5 |   +2%  ||     4.35 |     4.28 |   -2%  |  0.0000 |
 | TPC-H 06 ||      3.2 |     3.2 |   +0%  ||   315.38 |   314.31 |   -0%  |  0.0009 |
 | TPC-H 07 ||     59.2 |    59.4 |   +0%  ||    16.88 |    16.82 |   -0%  |  0.7267 |
 | TPC-H 08 ||     76.8 |    77.8 |   +1%  ||    13.02 |    12.85 |   -1%  |  0.0039 |
 | TPC-H 09 ||    521.8 |   531.7 |   +2%  ||     1.92 |     1.88 |   -2%  |  0.0000 |
 | TPC-H 10 ||    250.2 |   254.9 |   +2%  ||     4.00 |     3.92 |   -2%  |  0.0616 |
-| TPC-H 11 ||      8.5 |     9.0 |   +6%  ||   117.23 |   110.85 |   -5%  |  0.0000 |
 | TPC-H 12 ||     43.8 |    44.1 |   +1%  ||    22.82 |    22.67 |   -1%  |  0.0000 |
 | TPC-H 13 ||    359.5 |   354.7 |   -1%  ||     2.78 |     2.82 |   +1%  |  0.0000 |
 | TPC-H 14 ||     21.5 |    22.1 |   +3%  ||    46.41 |    45.15 |   -3%  |  0.0000 |
 | TPC-H 15 ||      7.7 |     7.7 |   -0%  ||   129.83 |   130.44 |   +0%  |  0.0000 |
 | TPC-H 16 ||     91.4 |    91.8 |   +0%  ||    10.94 |    10.90 |   -0%  |  0.1277 |
 | TPC-H 17 ||     19.4 |    19.4 |   +0%  ||    51.63 |    51.51 |   -0%  |  0.2131 |
-| TPC-H 18 ||    696.3 |   760.7 |   +9%  ||     1.44 |     1.31 |   -8%  |  0.0000 |
 | TPC-H 19 ||     30.6 |    31.1 |   +1%  ||    32.64 |    32.19 |   -1%  |  0.0000 |
 | TPC-H 20 ||     16.5 |    16.7 |   +1%  ||    60.59 |    59.75 |   -1%  |  0.0000 |
 | TPC-H 21 ||    417.4 |   419.0 |   +0%  ||     2.40 |     2.39 |   -0%  |  0.4359 |
 | TPC-H 22 ||     53.0 |    51.8 |   -2%  ||    18.85 |    19.32 |   +2%  |  0.0000 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Sum      ||   3889.1 |  3960.0 |   +2%  ||          |          |        |         |
 | Geomean  ||          |         |        ||          |          |   -1%  |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
```